frontend/packages/dev-console/src/components/deployments/deployment-strategy/DeploymentStrategySection.tsx

```diff
@@ -59,16 +59,18 @@ const DeploymentStrategySection: React.FC<DeploymentStrategySectionProps> = ({
         ..._.merge(strategyDefaultValues, deploymentStrategy),
         type: value,
       };
+      const oldType = initialValues.formData.deploymentStrategy.type;
       initialValues.formData.deploymentStrategy = strategyData;
+      initialValues.formData.deploymentStrategy.type = oldType;
```

Comment:
I think you won't need to set the initial values like this, 'cause we are already doing 
`setFieldValue('formData.deploymentStrategy', strategyData);`

Test the code after removing the usage of "initialValues" in this callback func